plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/DirectoryListingFilter.java

```diff
@@ -82,6 +83,18 @@ private TrinoFileStatus findNextElement()
         return null;
     }
 
+    private static boolean areLocationsEquivalent(Location location, Location other)
+    {
+        return location.equals(other) ||
+                (endsWithSlash(other) && !endsWithSlash(location) && Objects.equals(location + "/", other.toString())) ||
+                (!endsWithSlash(other) && endsWithSlash(location) && Objects.equals(location.toString(), other + "/"));
```

Comment:
Why do we need this?

Reply:
i think the base of the problem is that this doesn't hold

```
assertParentDirectory("s3://bucket/foo", Location.of("s3://bucket/"));
```

(using assertions from TestLocation)

Reply:
We verify here whether

`s3://bucketname` is equivalent with `s3://bucketname/`

Reply:
i think this solves also listing when table location has a trailing slash, right?
cc @homar @pajaks 

@findinpath 
please cherry pick test portion of https://github.com/trinodb/trino/pull/17586.
can you also add a test case where hive table is created with location that has a trailing slash?

yes, as it is this needs a comment
however, i would rather want `assertParentDirectory("s3://bucket/foo", Location.of("s3://bucket/"))` to be true
fixing top-of-the-bucket case alone won't solve listings when table location has a trailing slash.

Reply:
> i think this solves also listing when table location has a trailing slash, right?

@findepi  `TestDeltaLakeMinioAndHmsConnectorSmokeTest#testVacuumWithTrailingSlash` runs successfully now.

cc @pajaks